test/ecto/adapters/sqlite3_test.exs

```diff
@@ -3,6 +3,14 @@ defmodule Ecto.Adapters.SQLite3Test do
 
   alias Ecto.Adapters.SQLite3
 
+  @uuid_regex ~R/^[[:xdigit:]]{8}\b-[[:xdigit:]]{4}\b-[[:xdigit:]]{4}\b-[[:xdigit:]]{4}\b-[[:xdigit:]]{12}$/
+
+  setup do
+    on_exit(fn ->
+      Application.put_env(:ecto_sqlite3, :binary_id_type, :string)
+    end)
+  end
```

Comment:
Probably better to capture the original value and then restore it back after each test.

```suggestion
  setup do
    original_binary_id_type = Application.get_env(:ecto_sqlite3, :binary_id_type)
    on_exit(fn ->
      Application.put_env(:ecto_sqlite3, :binary_id_type, original_binary_id_type)
    end)
  end
```